### ISSUE TITLE
iOS version of the Commons app - market share

### ISSUE THREAD

**Issue by OpenGreenStreet** [commons-app/apps-android-commons#5828]
### What is the user problem or growth opportunity you want to see solved?

From my point of view, it would be good if there were an iOS version of the Commons app in the future, as one third of all smartphones (in Germany) are iOS smartphones.

### How do you know that this problem exists today? Why is this important?

This means that around 30 percent of all potential users of the commons app have no chance of using it.

### Who will benefit from it?

All users who own an iOS smartphone.

### Anything else you would like to add?

https://de.statista.com/statistik/daten/studie/256790/umfrage/marktanteile-von-android-und-ios-am-smartphone-absatz-in-deutschland/

**Comment by RitikaPahwa4444** [comment]
The app is now maintained by volunteers, so it might be hard to achieve. There used to be an active discussion about this when we used to receive grants: https://github.com/commons-app/apps-android-commons/issues/5060. We've seen that folks are interested in migrating the app to Jetpack Compose, so using Compose Multiplatform might be possible if we're able to gain enough support🙂

**Comment by misaochan** [comment]
Hi @OpenGreenStreet , we have always wanted to expand to iOS, however historically WMF Product has disagreed with it. That was a long time ago, however (around 3 years ago if I recall correctly), so it might be worth reopening the discussion again if folks are interested. If WMF does not wish to host the iOS app on their account, we can still self-publish, but user acquisition will be difficult.

**Comment by OpenGreenStreet** [comment]
@misaochan , @RitikaPahwa4444 , Thank you for your feedback. I had already thought of something like that and I can also imagine how difficult it would be in terms of content and resources. I just wanted to raise the issue again.

I have no idea, but would there perhaps be a way to get funding from the Wikimedia Foundation or Wikimedia Deutschland for such a project?

**Comment by misaochan** [comment]
> I have no idea, but would there perhaps be a way to get funding from the Wikimedia Foundation or Wikimedia Deutschland for such a project?

@OpenGreenStreet Previously, we were supported by WMF grants. I took a peek at those recently, and it seems like the General Support grant does not take software projects anymore. There was supposed to be a WMF Technology Fund for software projects, but it never got off the ground and is now considered ["permanently placed on hold"](https://meta.wikimedia.org/wiki/Grants_talk:Start#Timeline_for_%22Wikimedia_Technology_Fund%22). To be fair, I haven't talked to the program officers involved to see if there is any way around this, but just based on the official documentation, it seems like WMF is out.

Re: WM Deutschland, that is an interesting suggestion. I don't think we have collaborated with WMD before, would you have any suggestions on how to start if we were to go about that? :)

Of course, all of this depends on the other maintainers' opinions as well.

**Comment by OpenGreenStreet** [comment]
@misaochan , Thanks, I'll see if I can find a contact person.

**Comment by misaochan** [comment]
@OpenGreenStreet Fantastic, thank you! :) If you'd like, please feel free to CC me in your email to them. Personally, I've always been keen on making an iOS app happen.

**Comment by OpenGreenStreet** [comment]
@misaochan , what I have been able to achieve so far is an interesting collection of links documenting past efforts in this matter:

* https://meta.wikimedia.org/wiki/Community_Wishlist_Survey_2021/Archive/Create_Commons_iOS_app
* https://meta.wikimedia.org/wiki/Community_Wishlist/Wishes/IOS_Commons_App
* https://meta.wikimedia.org/wiki/Grants:Project/Commons_app/Commons_app_Android_v4


* https://meta.wikimedia.org/wiki/Grants:Start/de

**Comment by OpenGreenStreet** [comment]
@misaochan , hi have written to Wikimedia Deutschland and also put you on CC.

**Comment by misaochan** [comment]
@OpenGreenStreet Thank you very much for that. :) Yes, in the past WMF made it quite clear to me that they did not wish to fund an iOS app (the 3rd link that you posted was my grant proposal in 2020, which was declined and they asked me to resubmit another one with Android improvements instead). So IMO if it is ever going to happen, it will have to be with external funding, e.g. from a chapter. Looking forward to WMD's response.

**Comment by OpenGreenStreet** [comment]
@misaochan Thanks for your feedback and yes, let's wait and see what WMD replies. I would very much like there to be a solution.

**Comment by OpenGreenStreet** [comment]
By the way: the StreetComplete app, which I also like very much, is planning an iOS version:

https://github.com/streetcomplete/StreetComplete/issues/5421

**Comment by misaochan** [comment]
Update for anyone who is following: WMDE didn't get back to @OpenGreenStreet and I. However, considering that it's been 4 years, I think reopening the discussion with WMF is not a bad idea. So I reached out to a couple of my old contacts at WMF and we'll see what becomes of it.

**Comment by nicolas-raoul** [comment]
Deciders won't approve if we don't show them some clear quantified positive impact based on Wikimedia's strategic goals.

Would anyone be able to compute some statistics about the Android app, for the current year or past 12 months?
- Proportion of pictures uploaded via the app.
- Proportion of pictures in use, versus non-app pictures.
- Same for geolocalized pictures.
- Same for pictures with at least one depiction or non-technical category.
- Same for geolocalized pictures in "global south".
- Other statistics ideas welcome.

**Comment by misaochan** [comment]
> Proportion of pictures uploaded via the app.
> Proportion of pictures in use, versus non-app pictures.

I can get numbers for this on the app's side, but not sure how to obtain the total Commons number to calculate the proportion.

**Comment by OpenGreenStreet** [comment]
Could it perhaps also be useful to look at the use of Commons images in the OSM project.

See:
* https://commons.wikimedia.org/wiki/Commons:Files_used_on_OpenStreetMap
* https://commons.wikimedia.org/wiki/User:Usage_Bot

**Comment by OpenGreenStreet** [comment]
Just an idea …
 
#5924

**Comment by misaochan** [comment]
Just wanted to provide an update on this. I had a chat with the Director of Product recently about the app. In short, they are in the early stages of their discussion/strategy phase, and they haven't quite decided what direction they want to go with mobile Commons uploads in general. They don't seem to be **against** the creation of an iOS Commons app (which is an improvement over the previous status quo!), however, receiving funding from them doesn't seem to be in the books for the near future.

I'll provide further updates if I hear back from them again.

**Comment by OpenGreenStreet** [comment]
@misaochan 

Thanks for your comment: you're right of course, the number of donors won't be overwhelming and the sums probably won't be either. But certainly better than nothing... (In German they say: “Kleinvieh macht auch Mist.” (Small cattle also make dung.) :-)

PS: I had thought of 5 to 10 dollars per month for myself. Why shouldn't we contributors to Commons also make sure that we have/keep good tools.

**Comment by nylki** [comment]
(accidentally posted earlier in here: https://github.com/commons-app/apps-android-commons/issues/5924)
Heya, just a general PSA:
I have been actively working on a new native iOS App for Commons with a similar feature set to the Android Commons app since ca. October 2024.

There is no test version yet nor public code yet, but it will be open source with a FOSS license. I plan to have a first semi-public test version ready in March, 2025. Basic functionality like registration, login, and uploading and image searching already works on my internal branch and I am positive to have a first public release (AppStore or Testflight) mid-end next year. I don't receive any funding for that as of now, but have applied to NLNet and consider applying for a Wikimedia Rapid Grant once I made my repo public.

**Comment by OpenGreenStreet** [comment]
@nylki
My point is that a project like this is not dependent on a single person. (Otherwise I see the risk that there will suddenly be no more support. I mean, there has been this problem before?)

**Comment by nicolas-raoul** [comment]
@ny

Fantastic!

Do not hesitate to make your code public already, "release early release often". 🙂

If you are interested, WMF would probably accept it in their official App Store account once stable, it would have good visibility there. When I moved the app from my personal Android Play Store account to WMF's account, the increasez in downloads was noticeable.

Are you using some multiplatform framework, by any chance? If Jetpack Compose, we should strive towards converging and eventually merging if that's OK with everyone. 🙂

**Comment by misaochan** [comment]
@nylki That's exciting news! :) Please let us know when you have made the repo public so we can check it out. 

Rapid Grants are 5000 USD at maximum and I believe they are also contingent on it being a "one and done" deal (they don't want you to apply repeatedly for the same project). Interesting to hear about NLNet, do you know if it's only for Europeans?

All the best, and we're happy to help if you need anything!

**Comment by misaochan** [comment]
> Are you using some multiplatform framework, by any chance? If Jetpack Compose, we should strive towards converging and eventually merging if that's OK with everyone. 🙂

When I last talked to the WMF iOS engineering team, they required the proposed app to be native iOS in order for them to put it on the WMF account. That was a long time ago, and I think the person who required it isn't working for WMF anymore, so it might be worth getting their opinion again.

**Comment by nylki** [comment]
Thank you for the warm words and excitement from you folks, much appreciated! :slightly_smiling_face: 

However, I'd also like to manage your expectation a bit. Although I am determined, it is still early days, so to speak.
After seeing the threads here and in the Commons-VillagePump, I thought I'd give you a small heads-up. Also thank you for offering support if I need help, I may come back to that offer, especially when it comes to API usage of some of the more elusive queries :smile: 

> When I last talked to the WMF iOS app team, they required the proposed app to be native iOS in order for them to consider putting it on the WMF account. We had made a demo prototype in Flutter that we showed them, but they declined to host a multiplatform app. That was a long time ago, though, so it might be worth getting their opinion again if @nylki wants to go the multiplatform route.


My app is written in Swift/SwiftUI, so it will be a native platform app. I did consider the multiplatform route, but decided against it, as I would not have been able to produce solid results in a reasonable time if I had to learn a new framework. I developed with Swift and SwiftUI in a professional context and feel comfortable with it (and do also enjoy working with it, which is an important part when it comes to projects like these).
One advantage will also be, that porting/adapting to a native MacOS, tvOS and IPad version in the future will be relatively straight forward.

> Rapid Grants are 5000 USD at maximum and I believe they are also contingent on it being a "one and done" deal (they don't want you to apply repeatedly for the same project). Interesting to hear about NLNet, do you know if it's only for Europeans?
Ah that is interesting! Thanks for the heads-up.
About NLNet: They are based and funded in the EU, but they write that they do consider worldwide applications if it fits their goals: https://nlnet.nl/commonsfund/faq/#geolimitations

**Comment by OpenGreenStreet** [comment]
@misaochan, @nicolas-raoul : 
Does the following link possibly help?
* https://meta.wikimedia.org/wiki/Grants:Programs/Wikimedia_Community_Fund/General_Support_Fund

**Comment by misaochan** [comment]
Hi @OpenGreenStreet , I looked at that before. Unfortunately it says:

> Examples of proposals we do not fund
> Proposals that fit certain criteria are not eligible for review in the Wikimedia Community Fund:
> Proposals that are primarily focused on research activities should be submitted to the [Wikimedia Research Fund](https://meta.wikimedia.org/wiki/Special:MyLanguage/Grants:Programs/Wikimedia_Research_%26_Technology_Fund).
> **Proposals that are primarily focused on large-scale software development are not eligible for review.** Smaller-scale software development projects may be eligible for review in the [Rapid Fund](https://meta.wikimedia.org/wiki/Special:MyLanguage/Grants:Project/Rapid) pending review from [Community Resources](https://meta.wikimedia.org/wiki/Special:MyLanguage/Community_Resources) and [Technology department](https://www.mediawiki.org/wiki/Wikimedia_Technology) staff.

**Comment by misaochan** [comment]
@OpenGreenStreet Also, if you like you could try writing directly to https://meta.wikimedia.org/wiki/User:JTanner_(WMF) and https://meta.wikimedia.org/wiki/User:MMiller_(WMF). Regardless of what form of Wikimedia funding we pursue, it would eventually be brought up for their approval, so feedback from Wikimedians might help. I've talked to them already, but they haven't gotten back to me (they might be off for the Christmas and New Year holidays).

**Comment by OpenGreenStreet** [comment]
Hi @misaochan , I think it would be better if someone from the developers did this. (I have no experience and my English skills are only moderate :-) If I can be helpful with short comments or with votes, then very gladly.

**Comment by nylki** [comment]
Hey folks!
Although it is in fact not march anymore, but I did make progress on my iOS app and made the repo public now: https://github.com/nylki/CommonsFinder

There you can also find some screen recordings of what the current state looks like.

If you are interested in trying it out for yourself and report crashes and give feedback, here is the TestFlight invite: https://testflight.apple.com/join/15KtE2Mn
The invite is currently limited to 25 testers and I am sharing it in a few other places, so please don't share to too much around, just yet 🙂

**Comment by OpenGreenStreet** [comment]
@nylki : Oh, that sounds great. I'll keep my fingers crossed that it works out this time.

**Comment by misaochan** [comment]
Hi @nylki ! Just following up on how things are going with the iOS app? I checked out the repo recently and IMO the upload screenshots look fantastic. :)

Also, I was wondering if there are any plans to implement the "Nearby places that need photos" feature in the near future? I did see a map in the demo, but it seems to display existing Commons photos if I understand correctly.

**Comment by nylki** [comment]
> Hi [@nylki](https://github.com/nylki) ! Just following up on how things are going with the iOS app? I checked out the repo recently and IMO the upload screenshots look fantastic. :)
> 

Thank you, @misaochan, much appreciated!

> Also, I was wondering if there are any plans to implement the "Nearby places that need photos" feature in the near future? I did see a map in the demo, but it seems to display existing Commons photos if I understand correctly.

As a matter of fact I am currently working on a bigger set of changes for the map, to make it more useful and intuitive overall. While that exact feature won't make it into that changeset for the next 1-3 weeks, I definetely do want to get that in the near future afterwards. (and I do really want to use that feature myself on photo trips, ...so the motiviation is there :D)

**Comment by misaochan** [comment]
Hi @nylki , will you be attending the Wikimedia Hackathon in Milan by any chance?

**Comment by nylki** [comment]
> Hi [@nylki](https://github.com/nylki) , will you be attending the Wikimedia Hackathon in Milan by any chance?

Heya @misaochan, I can't make it to the Hackathon unfortunately, I am already on a vacation at that time.

Ah btw. to keep you in the loop as well:
I plan to publish CommonsFinder this year on the AppStore, most likely under my own account with the current name initially. Most core features I want the app to have are in there now and getting stable.

There is still a blocking issue with the recent rate-limiting changes, hitting thumb/resized images with http-429 when multiple users on different devices use the app at the same time (not sure if this is also an issue for the Android Commons App, but that might be a topic for a different ticket/discussion :D ).
Ideally I'd like to also have OAuth2 in before release, but there are still some phabricator backend tickets that are needed for that to work in mobile apps.

**Comment by misaochan** [comment]
Hey @nylki , that's great news! :) Let us know when you've launched and we can add your App Store link to our wiki and such.

Yeah feel free to create an issue for the rate-limiting changes if a quick search doesn't bring up anything! I'm not too sure about the recent changes but I'm sure someone will be able to help haha.

Pretty sure we don't have OAuth2 either.

**Comment by nylki** [comment]
> Hey [@nylki](https://github.com/nylki) , that's great news! :) Let us know when you've launched and we can add your App Store link to our wiki and such.

Will do! :)

**Comment by misaochan** [comment]
I just noticed that there are a couple of Community Wishlist items relevant to this:

https://meta.wikimedia.org/wiki/Community_Wishlist/W171
https://meta.wikimedia.org/wiki/Community_Wishlist/W252
